pkg/client/fileCheckpointer.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2022 IBM All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package client
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+)
+
+type FileCheckpointer struct {
+	path          string
+	blockNumber   uint64
+	transactionID string
+}
+
+type checkpointState struct {
+	BlockNumber   uint64 `json:"blockNumber"`
+	TransactionID string `json:"transactionId"`
+}
+
+func NewFileCheckpointer(path string) (*FileCheckpointer, error) {
+	fileCheckpointer := new(FileCheckpointer)
+	fileCheckpointer.path = path
+
+	err := fileCheckpointer.loadFromFile()
+	if err != nil {
+		return fileCheckpointer, err
+	}
+
+	err = fileCheckpointer.saveToFile()
+	return fileCheckpointer, err
+}
+
+func (c *FileCheckpointer) CheckpointBlock(blockNumber uint64) error {
+	c.updateState(blockNumber+1, "")
+	return c.saveToFile()
+}
+
+func (c *FileCheckpointer) CheckpointTransaction(blockNumber uint64, transactionID string) error {
+	c.updateState(blockNumber, transactionID)
+	return c.saveToFile()
+}
+
+func (c *FileCheckpointer) CheckpointChaincodeEvent(event *ChaincodeEvent) error {
+	err := c.CheckpointTransaction(event.BlockNumber, event.TransactionID)
+	return err
+}
+
+func (c *FileCheckpointer) BlockNumber() uint64 {
+	return c.blockNumber
+}
+
+func (c *FileCheckpointer) TransactionID() string {
+	return c.transactionID
+}
+
+func (c *FileCheckpointer) State() *FileCheckpointer {
+	return &FileCheckpointer{
+		blockNumber:   c.blockNumber,
+		transactionID: c.transactionID,
+	}
+}
```

Comment:
Something looks wrong here. This method shouldn't be public and I don't think it should be here at all. It allows a FileCheckpointer to return an invalid FileCheckpointer (missing a `path`).